src/Lykke.Bil2.Contract/BlocksReader/Events/BalanceChange.cs

```diff
@@ -78,12 +108,15 @@ public class BalanceChange
             if (string.IsNullOrWhiteSpace(assetId))
                 throw new ArgumentException("Should be not empty string", nameof(assetId));
 
-            if (string.IsNullOrWhiteSpace(address))
-                throw new ArgumentException("Should be not empty string", nameof(address));
+            if (address != null && string.IsNullOrWhiteSpace(address))
```

Comment:
This is always false
address!=null && address ==null

Reply:
No, address can be null, but after implicit conversion to the string it can be empty or whitespace string. Actually, there should be a constraint in the Asset type. I'll add such constraint for all value types later and then remove this wired check.

Reply:
There is !string.IsNulltOrWhitespace constraint for all value types which can be implicitly converted to the string. All checks are fixed